### ISSUE TITLE
Loading icon (and probably other animated icons) cause high CPU usage

### ISSUE THREAD

**Issue by jwarkentin** [microsoft/vscode#140479]
This is really a duplicate of issue #127235 which was closed as a duplicate of issue #126197 even though they are two different issues. The solution to #126197 only allows the option of changing a setting to remove the loading icon for the terminal (and it was all about the distraction, not the high CPU usage) but if the loading icon is used by anything else then the problem persists.

As an example, see [this](https://github.com/spmeesseman/vscode-taskexplorer/issues/93) issue with the Task Explorer extension which was closed since it's an issue with VSCode. For many more examples of related problems just Google something like "vscode animated icons high cpu". People work around it by trying to disable animated icons but that's not always possible. With the Task Explorer extension I can only work around it by collapsing the tree while I have an active long-running task (like `webpack --watch`).

Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.64.0-insider
- OS Version: Ubuntu 21.10

Steps to Reproduce:

1. Do anything that renders the built-in `loading.svg` icon
2. Observe that CPU usage spikes and remains high (~25% of one core for me) while it is animating the spinner

**Comment by jwarkentin** [comment]
So I've been playing around a bit today and wanted to mention one observation. Normally, when a spin animation is rendering, the process monitor shows one process (the `code --type=renderer` one) consuming ~28% CPU and another (the `code --type=gpu-process` one) using ~22% CPU. However, if I start VSCode with the `--enable-gpu-rasterization` flag then both processes tend to hang around between 18-22% CPU.

That's still not great but it's a noticeable improvement. I tried adding the `--enable-zero-copy` flag to that but doesn't make a difference like I thought it would. I'm not sure what exactly that means for the source of the problem since the bulk of the animation performance penalty still exists but thought it should be mentioned.

**Comment by jwarkentin** [comment]
Until now I've only looked at this using X11 on Ubuntu. However, I've been trying to switch over to Wayland and was just testing with that. There's an intriguing reversal of CPU load. No matter what options I pass (with / without ozone, GPU rasterization, etc...), the GPU process consumes ~46-52% CPU and the renderer process ends up taking ~10-12% CPU while animating. If I set the ozone platform backend to Wayland then VSCode won't even start so it clearly must be using the x11 backend and running with XWayland.

The way that the renderer process's CPU usage scales inversely to the GPU process is telling. The GPU layer is going to be less efficient under XWayland since it has to be translated through the XServer before being passed to Wayland. That means the renderer process is likely only taking less CPU because it's spending more time waiting on the GPU process. So if the renderer process were optimized to do less of whatever it's doing calling the GPU process to render the animation, or do it more efficiently somehow, then it seems the whole thing would be more performant and both processes would be less busy.
